src/Type/Php/InArrayFunctionTypeSpecifyingExtension.php

```diff
@@ -65,27 +84,69 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 			);
 		}
 
+		// If e.g. needle is 'a' and haystack non-empty-array<int, 'a'> we can be sure that this always evaluates to true
+		// Belows HasOffset::isSuperTypeOf cannot deal with that since it calls ArrayType::hasOffsetValueType and that returns maybe at max
+		if ($needleType instanceof ConstantScalarType && $arrayType->isIterableAtLeastOnce()->yes() && $arrayValueType->equals($needleType)) {
+			return $specifiedTypes;
+		}
+
```

Comment:
I also dislike this edge case, maybe it is possible to adapt `HasOffset::isSuperTypeOf` somehow to figure this out.. Any ideas?